UpdateDotnetRuntime.ps1

```diff
@@ -0,0 +1,130 @@
+[CmdletBinding()]
+param (
+)
+
+function Update-GlobalJson([string] $Version) {
+    $psGlobalJsonPath = "$PSScriptRoot/global.json"
+    $psGlobalJson = Get-Content -Path $psGlobalJsonPath -Raw | ConvertFrom-Json
+    $psGlobalJson.sdk.version = $Version
+    $psGlobalJson | ConvertTo-Json | Out-File -FilePath $psGlobalJsonPath -Force
+}
+
+function Update-PackageVersion {
+
+    class PkgVer {
+        [string] $Name
+        [string] $Version
+        [string] $NewVersion
+        [string] $Path
+
+        PkgVer($n, $v, $nv, $p) {
+            $this.Name = $n
+            $this.Version = $v
+            $this.NewVersion = $nv
+            $this.Path = $p
+        }
+    }
+
+    $skipModules = @(
+        "NJsonSchema"
+        "Markdig.Signed"
+        "PowerShellHelpFiles"
+        "Newtonsoft.Json"
+        "Microsoft.ApplicationInsights"
+        "Microsoft.Management.Infrastructure"
+        "Microsoft.PowerShell.Native"
+        "Microsoft.NETCore.Windows.ApiSets"
+    )
+
+    $packages = [System.Collections.Generic.Dictionary[[string], [PkgVer]]]::new()
+
+    Get-ChildItem -Path "$PSScriptRoot\src\" -Recurse -Filter "*.csproj" -Exclude 'PSGalleryModules.csproj' | ForEach-Object {
+        $prj = [xml] (Get-Content $_.FullName -Raw)
+        $pkgRef = $prj.Project.ItemGroup.PackageReference
+
+        foreach ($p in $pkgRef) {
+            if ($null -ne $p -and -not $skipModules.Contains($p.Include)) {
+                if (-not $packages.ContainsKey($p.Include)) {
+                    $packages.Add($p.Include, [PkgVer]::new($p.Include, $p.Version, $null, $_.FullName))
+                }
+            }
+        }
+    }
+
+    $versionPattern = (Get-Content "$PSScriptRoot\DotnetRuntimeMetadata.json" | ConvertFrom-Json).sdk.packageVersionPattern
+
+    $packages.GetEnumerator() | ForEach-Object {
+        $pkgs = Find-Package -Name $_.Key -AllVersions -AllowPreReleaseVersions -Source 'dotnet5'
+
+        $version = $_.Value.Version
+
+        foreach ($p in $pkgs) {
+            if ($p.Version -like "$versionPattern*") {
+                if ([System.Management.Automation.SemanticVersion] ($version) -lt [System.Management.Automation.SemanticVersion] ($p.Version)) {
+                    $_.Value.NewVersion = $p.Version
+                    break
+                }
+            }
```

Comment:
I think it is more simple to add new property in PowerShell.Common.props file and use the property in csproj files.

Reply:
The versions for different packages can be different. We would need to add a property per package. Also the intention of this script is to be run manually before opening a PR to update .NET.